nixos/modules/services/web-apps/hatsu.nix

```diff
@@ -0,0 +1,101 @@
+{
+  lib,
+  pkgs,
+  config,
+  ...
+}:
+let
+  cfg = config.services.hatsu;
+in
+{
+  meta.doc = ./hatsu.md;
+  meta.maintainers = with lib.maintainers; [ kwaa ];
+
+  options.services.hatsu = {
+    enable = lib.mkEnableOption "Self-hosted and fully-automated ActivityPub bridge for static sites";
+
+    package = lib.mkPackageOption pkgs "hatsu" { };
+
+    dataDir = lib.mkOption {
+      type = lib.types.str;
+      default = "/var/lib/hatsu";
+      description = "Data directory for hatsu.";
+    };
+
+    settings = lib.mkOption {
+      type = lib.types.submodule {
+        freeformType =
+          with lib.types;
+          attrsOf (
+            nullOr (oneOf [
+              str
+              path
+              package
+            ])
+          );
+        options = {
+          HATSU_DATABASE_URL = lib.mkOption {
+            type = lib.types.str;
+            default = "sqlite://${cfg.dataDir}/hatsu.sqlite?mode=rwc";
+            defaultText = "sqlite://\${config.services.hatsu.dataDir}/hatsu.sqlite?mode=rwc";
+            example = "postgres://username:password@host/database";
+            description = "Database URL.";
+          };
+
+          HATSU_DOMAIN = lib.mkOption {
+            type = lib.types.str;
+            description = "The domain name of your instance (eg 'hatsu.local').";
+          };
+
+          HATSU_LISTEN_HOST = lib.mkOption {
+            type = lib.types.str;
+            default = "127.0.0.1";
+            description = "Host where hatsu should listen for incoming requests.";
+          };
+
+          HATSU_LISTEN_PORT = lib.mkOption {
+            type = lib.types.port;
```

Comment:
```suggestion
            type = lib.types.port;
            apply = toString;
```

Reply:
Fixed, I also changed the value of `freeformType`